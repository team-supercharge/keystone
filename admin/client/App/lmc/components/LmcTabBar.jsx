import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Link } from 'react-router'

export class LmcTabBar extends Component {
    constructor (props) {
        super(props);
    }

    isActive = (url) => {
        const pathname = this.props.location.pathname
        if (pathname) {
            return pathname.match(url) ? true : false
        }
    }
    
    render () {
        const { items, resourceUrl } = this.props
        const baseUrl = `${Keystone.adminPath}/${resourceUrl}`

        return (
            <nav className='secondary-navbar' style={styles.navbar}>
                <div style={styles.wrapper}>
                    <ul className="app-nav app-nav--secondary app-nav--left">
                        { items.map((item, index) => {
                            return this.isActive(item.url)
                                ? <li 
                                    className='lmc-secondary-nav-link'
                                    key={index}
                                    style={{ ...styles.activeItem, ...styles.item }}
                                >
                                    <Link 
                                        className='lmc-secondary-nav-link'
                                        to={`${baseUrl}/${item.url}`}
                                    >
                                        {item.label}
                                    </Link>
                                </li>
                                : <li 
                                    className='lmc-secondary-nav-link'
                                    key={index}
                                    style={styles.item}>
                                    <Link
                                        className='lmc-secondary-nav-link'
                                        to={`${baseUrl}/${item.url}`}
                                    >
                                        {item.label}
                                    </Link>
                                </li>
                        }) }
                    </ul>
                </div>
            </nav>
        )
    }
}

const styles = {
    activeItem: {
        borderBottom: '3px solid #e65d78',
    },
    item: {
        boxSizing: 'border-box',
        flex: '1',
        textAlign: 'center',
        padding: '7px 10px 9px 10px',
    },
    navbar: {
        height: '58px',
        backgroundColor: 'white',
        paddingLeft: 20,
        paddingRight: 50,
    },
    wrapper: {
        display: 'flex',
    }
}

LmcTabBar.propTypes = {
    resourceUrl: PropTypes.string,
    items: PropTypes.array
}

LmcTabBar.defaultProps = {
    resourceUrl: 'residents',
    items: [{ label: 'Profile', url: 'profile' }],
}

const mapStateToProps = (state) => {
    return {
        selectedResident: state.residents.selectedResident,
    }
}

export default connect(mapStateToProps)(LmcTabBar)