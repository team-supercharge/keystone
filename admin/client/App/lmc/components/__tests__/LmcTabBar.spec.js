import React from 'react'
import { shallow } from 'enzyme'
import { LmcTabBar } from '../LmcTabBar.jsx'
import { Link } from 'react-router'

describe('LmcResidentsNavbar', () => {
    let wrapper
    let items
    let resourceUrl
    let location
    let savedKeystone

    beforeAll(() => {
        savedKeystone = global.Keystone
        global.Keystone = {
            adminPath: '/admin'
        }
        location = {
            pathname: 'testPathname'
        }
        items = [
            { label: 'label1', url: 'url1' },
            { label: 'label2', url: 'url2' }
        ]
        resourceUrl = 'testResourceUrl'
    })

    beforeEach(() => {
        wrapper = shallow(
            <LmcTabBar
                location={location}
                items={items}
                resourceUrl={resourceUrl}
            />
        )
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('renders a navigation element', () => {
        expect(wrapper.find('nav').length).toEqual(1)
    })

    it('renders an unordered list', () => {
        expect(wrapper.find('ul').length).toEqual(1)
    })

    it('renders links to the correct urls', () => {
        const firstLink = wrapper.find(Link).first()
        const secondLink = wrapper.find(Link).at(1)

        expect(firstLink.props().to).toEqual('/admin/testResourceUrl/url1')
        expect(secondLink.props().to).toEqual('/admin/testResourceUrl/url2')
    })

    afterAll(() => {
        global.Keystone = savedKeystone
    })
})