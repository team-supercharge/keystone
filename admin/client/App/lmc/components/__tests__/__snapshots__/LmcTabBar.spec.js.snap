// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcResidentsNavbar renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <nav
      className="secondary-navbar"
      style={
        Object {
          "backgroundColor": "white",
          "height": 41.5,
          "paddingLeft": 20,
          "paddingRight": 50,
        }
      }
    >
      <ul
        className="app-nav app-nav--secondary app-nav--left"
      >
        <li
          className="lmc-secondary-nav-link"
          style={
            Object {
              "boxSizing": "border-box",
              "flex": "1",
              "textAlign": "center",
            }
          }
        >
          <Link
            className="lmc-secondary-nav-link"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/admin/testResourceUrl/url1"
          >
            label1
          </Link>
        </li>
        <li
          className="lmc-secondary-nav-link"
          style={
            Object {
              "boxSizing": "border-box",
              "flex": "1",
              "textAlign": "center",
            }
          }
        >
          <Link
            className="lmc-secondary-nav-link"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/admin/testResourceUrl/url2"
          >
            label2
          </Link>
        </li>
      </ul>
    </nav>
  </div>,
  "nodes": Array [
    <div>
      <nav
        className="secondary-navbar"
        style={
          Object {
            "backgroundColor": "white",
            "height": 41.5,
            "paddingLeft": 20,
            "paddingRight": 50,
          }
        }
      >
        <ul
          className="app-nav app-nav--secondary app-nav--left"
        >
          <li
            className="lmc-secondary-nav-link"
            style={
              Object {
                "boxSizing": "border-box",
                "flex": "1",
                "textAlign": "center",
              }
            }
          >
            <Link
              className="lmc-secondary-nav-link"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/admin/testResourceUrl/url1"
            >
              label1
            </Link>
          </li>
          <li
            className="lmc-secondary-nav-link"
            style={
              Object {
                "boxSizing": "border-box",
                "flex": "1",
                "textAlign": "center",
              }
            }
          >
            <Link
              className="lmc-secondary-nav-link"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/admin/testResourceUrl/url2"
            >
              label2
            </Link>
          </li>
        </ul>
      </nav>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LmcTabBar
        items={
          Array [
            Object {
              "label": "label1",
              "url": "url1",
            },
            Object {
              "label": "label2",
              "url": "url2",
            },
          ]
        }
        location={
          Object {
            "pathname": "testPathname",
          }
        }
        resourceUrl="testResourceUrl"
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LmcTabBar {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "isActive": [Function],
        "props": Object {
          "items": Array [
            Object {
              "label": "label1",
              "url": "url1",
            },
            Object {
              "label": "label2",
              "url": "url2",
            },
          ],
          "location": Object {
            "pathname": "testPathname",
          },
          "resourceUrl": "testResourceUrl",
        },
        "refs": Object {},
        "renderLabel": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <nav
            className="secondary-navbar"
            style={
              Object {
                "backgroundColor": "white",
                "height": 41.5,
                "paddingLeft": 20,
                "paddingRight": 50,
              }
            }
          >
            <ul
              className="app-nav app-nav--secondary app-nav--left"
            >
              <li
                className="lmc-secondary-nav-link"
                style={
                  Object {
                    "boxSizing": "border-box",
                    "flex": "1",
                    "textAlign": "center",
                  }
                }
              >
                <Link
                  className="lmc-secondary-nav-link"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/admin/testResourceUrl/url1"
                >
                  label1
                </Link>
              </li>
              <li
                className="lmc-secondary-nav-link"
                style={
                  Object {
                    "boxSizing": "border-box",
                    "flex": "1",
                    "textAlign": "center",
                  }
                }
              >
                <Link
                  className="lmc-secondary-nav-link"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/admin/testResourceUrl/url2"
                >
                  label2
                </Link>
              </li>
            </ul>
          </nav>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <nav
            className="secondary-navbar"
            style={
              Object {
                "backgroundColor": "white",
                "height": 41.5,
                "paddingLeft": 20,
                "paddingRight": 50,
              }
            }
          >
            <ul
              className="app-nav app-nav--secondary app-nav--left"
            >
              <li
                className="lmc-secondary-nav-link"
                style={
                  Object {
                    "boxSizing": "border-box",
                    "flex": "1",
                    "textAlign": "center",
                  }
                }
              >
                <Link
                  className="lmc-secondary-nav-link"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/admin/testResourceUrl/url1"
                >
                  label1
                </Link>
              </li>
              <li
                className="lmc-secondary-nav-link"
                style={
                  Object {
                    "boxSizing": "border-box",
                    "flex": "1",
                    "textAlign": "center",
                  }
                }
              >
                <Link
                  className="lmc-secondary-nav-link"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/admin/testResourceUrl/url2"
                >
                  label2
                </Link>
              </li>
            </ul>
          </nav>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcTabBar
    items={
      Array [
        Object {
          "label": "label1",
          "url": "url1",
        },
        Object {
          "label": "label2",
          "url": "url2",
        },
      ]
    }
    location={
      Object {
        "pathname": "testPathname",
      }
    }
    resourceUrl="testResourceUrl"
  />,
}
`;
